leaflet-geoman.d.ts

```diff
@@ -26,6 +26,12 @@ declare module 'leaflet' {
     interface Layer {
         pm: PM.PMLayer;
     }
+    /**
+     * Extends built in leaflet layergroup.
+     */
+    interface LayerGroup {
+        pm: PM.PMLayerGroup;
+    }
```

Comment:
@ryan-morris does this works? I add the PMLayerGroup interface to the LayerGroup instead of the PMLayer interface

Reply:
This actually is creating an error at that line,

```
Interface 'LayerGroup<P>' incorrectly extends interface 'Layer'.
  Types of property 'pm' are incompatible.
    Type 'PMLayerGroup' is missing the following properties from type 'PMLayer': getShape, enableRotate, disableRotate, rotateEnabled, and 8 more.ts(2430)
```

It shouldn't be needed. `LayerGroup` in `@types/leaflet` inherits from `Layer` so `pm` should have already been available, like:

```
this.pmLayers = new LayerGroup([]);
this.pmLayers.pm.disable();
```

Reply:
I misread your point and realize why you did that. Sorry I didn't respond sooner on this. That _should_ be on LayerGroup but it is conflicting with the `Layer` declaration of `pm`. I'm trying to see if there's a way around there. Error message goes away if I take `pm: PM.PMLayer;` off of `Layer` , but that is not a valid solution.

Reply:
Are you sure? because LayerGroups needs different functions on the pm object then default Layers

Reply:
Question:: We have `interface PMLayer extends PMRotateLayer, PMEditLayer, PMDragLayer` , should any of the methods listed on that or the *Layer it extends, be available on a `LayerGroup`?

Reply:
No, layergroup does not need / use any of the layer interfaces. So extending is not working

Reply:
https://www.reddit.com/r/typescript/comments/e0krdu/overriding_interface_type_declarations/

>You can't do that, however you can do this:
type IBar = Omit<IFoo, 'test'> & {
    test: Partial<IFoo['test']>;
};